source/containerblob/identifier.go

```diff
@@ -0,0 +1,69 @@
+package containerblob
+
+import (
+	"log"
+	"strings"
+
+	"github.com/containerd/containerd/reference"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/solver/llbsolver/provenance"
+	"github.com/moby/buildkit/source"
+	srctypes "github.com/moby/buildkit/source/types"
+	digest "github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+type ImageBlobIdentifier struct {
+	Reference  reference.Spec
+	RecordType client.UsageRecordType
+	Filename   string
```

Comment:
Out of curiosity - might we want the ability to unpack a directory from an image blob?

For example, this works great for replacing sources that result in a single file, like HTTP, but what about for Git or Local? I guess we care less about Git, but there are some interesting use cases for capturing the Local directory, since it may not be reproducible between users (depending on the build process).

I suppose you would just be able to do this by having a single layered image with an empty config (though this would require an extra level of indirection).

Reply:
Maybe better would be for the policy to provide multi-vertex LLB instead of single op in this case. In that case, you could pair it with a `FileOp` that does the extraction.

I don't really care about local much because there is no checksum that tracks locals anyway. Git is interesting but it would really be only valuable if we can validate the original checksum based on the commit sha. If we have Git source with `sha1:abc` and then some targz blob that we claim is equivalent with targz checksum `sha256:def` there is no way to validate it either visually or by buildkit. So it looks like if we want to store snapshot of git source outside of the real git repository we need some special way to import it that still validates the original commit sha (and makes sure there are no extra files etc).